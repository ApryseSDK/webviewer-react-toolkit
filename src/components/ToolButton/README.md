# Component Information

A button designed to be used in small toolbars. This button would most likely
take an icon as its child.
