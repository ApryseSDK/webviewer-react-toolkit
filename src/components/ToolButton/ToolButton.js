'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.ToolButton = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var hooks_1 = require('../../hooks');
var Icon_1 = require('../Icon');
var IconButton_1 = require('../IconButton');
exports.ToolButton = (0, react_1.forwardRef)(function(_a, ref) {
  var className = _a.className,
    expandProps = _a.expandProps,
    children = _a.children,
    onClick = _a.onClick,
    buttonProps = tslib_1.__rest(_a, [
      'className',
      'expandProps',
      'children',
      'onClick',
    ]);
  var handleOnClick = (0, hooks_1.useOnClick)(onClick, {
    stopPropagation: true,
  });
  var isUserTabbing = (0, hooks_1.useAccessibleFocus)();
  var positionRight =
    (expandProps === null || expandProps === void 0
      ? void 0
      : expandProps.position) === 'right';
  var expandClass =
    expandProps === null || expandProps === void 0
      ? void 0
      : expandProps.className;
  var hasExpandProps = !!expandProps;
  var classes = (0, react_1.useMemo)(
    function() {
      var enabledObj = {
        'ui__toolButton--disabled': buttonProps.disabled,
        'ui__toolButton--tabbing': isUserTabbing,
        'ui__toolButton--expanded': hasExpandProps,
        'ui__toolButton--right': positionRight,
        'ui__toolButton--bottom': !positionRight,
      };
      var wrapper = (0, classnames_1.default)(
        'ui__base ui__toolButton',
        enabledObj,
      );
      var action = (0, classnames_1.default)(
        'ui__toolButton__action',
        enabledObj,
        className,
      );
      var expand = (0, classnames_1.default)(
        'ui__toolButton__expand',
        enabledObj,
        expandClass,
      );
      return { wrapper: wrapper, action: action, expand: expand };
    },
    [
      buttonProps.disabled,
      className,
      expandClass,
      hasExpandProps,
      isUserTabbing,
      positionRight,
    ],
  );
  return react_1.default.createElement(
    'div',
    { className: classes.wrapper },
    react_1.default.createElement(
      IconButton_1.IconButton,
      tslib_1.__assign({ disabled: buttonProps.disabled }, buttonProps, {
        className: classes.action,
        onClick: handleOnClick,
        ref: ref,
      }),
      children,
    ),
    expandProps
      ? react_1.default.createElement(
          IconButton_1.IconButton,
          tslib_1.__assign({ disabled: buttonProps.disabled }, expandProps, {
            className: classes.expand,
          }),
          react_1.default.createElement(Icon_1.Icon, { icon: 'ChevronDown' }),
        )
      : undefined,
  );
});
//# sourceMappingURL=ToolButton.js.map
