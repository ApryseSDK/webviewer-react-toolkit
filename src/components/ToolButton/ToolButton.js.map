{"version":3,"file":"ToolButton.js","sourceRoot":"","sources":["ToolButton.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAAwF;AACxF,qCAA6D;AAE7D,gCAA+B;AAC/B,4CAA2C;AAa9B,QAAA,UAAU,GAAG,IAAA,kBAAU,EAClC,UAAC,EAA6D,EAAE,GAAG;IAAhE,IAAA,SAAS,eAAA,EAAE,WAAW,iBAAA,EAAE,QAAQ,cAAA,EAAE,OAAO,aAAA,EAAK,WAAW,sBAA3D,mDAA6D,CAAF;IAC1D,IAAM,aAAa,GAAG,IAAA,kBAAU,EAAC,OAAO,EAAE,EAAE,eAAe,EAAE,IAAI,EAAE,CAAC,CAAC;IAErE,IAAM,aAAa,GAAG,IAAA,0BAAkB,GAAE,CAAC;IAE3C,IAAM,aAAa,GAAG,CAAA,WAAW,aAAX,WAAW,uBAAX,WAAW,CAAE,QAAQ,MAAK,OAAO,CAAC;IACxD,IAAM,WAAW,GAAG,WAAW,aAAX,WAAW,uBAAX,WAAW,CAAE,SAAS,CAAC;IAC3C,IAAM,cAAc,GAAG,CAAC,CAAC,WAAW,CAAC;IAErC,IAAM,OAAO,GAAG,IAAA,eAAO,EAAC;QACtB,IAAM,UAAU,GAAG;YACjB,0BAA0B,EAAE,WAAW,CAAC,QAAQ;YAChD,yBAAyB,EAAE,aAAa;YACxC,0BAA0B,EAAE,cAAc;YAC1C,uBAAuB,EAAE,aAAa;YACtC,wBAAwB,EAAE,CAAC,aAAa;SACzC,CAAC;QACF,IAAM,OAAO,GAAG,IAAA,oBAAU,EAAC,yBAAyB,EAAE,UAAU,CAAC,CAAC;QAClE,IAAM,MAAM,GAAG,IAAA,oBAAU,EAAC,wBAAwB,EAAE,UAAU,EAAE,SAAS,CAAC,CAAC;QAC3E,IAAM,MAAM,GAAG,IAAA,oBAAU,EAAC,wBAAwB,EAAE,UAAU,EAAE,WAAW,CAAC,CAAC;QAC7E,OAAO,EAAE,OAAO,SAAA,EAAE,MAAM,QAAA,EAAE,MAAM,QAAA,EAAE,CAAC;IACrC,CAAC,EAAE,CAAC,WAAW,CAAC,QAAQ,EAAE,SAAS,EAAE,WAAW,EAAE,cAAc,EAAE,aAAa,EAAE,aAAa,CAAC,CAAC,CAAC;IAEjG,OAAO,CACL,uCAAK,SAAS,EAAE,OAAO,CAAC,OAAO;QAC7B,8BAAC,uBAAU,qBACT,QAAQ,EAAE,WAAW,CAAC,QAAQ,IAC1B,WAAW,IACf,SAAS,EAAE,OAAO,CAAC,MAAM,EACzB,OAAO,EAAE,aAAa,EACtB,GAAG,EAAE,GAAG,KAEP,QAAQ,CACE;QACZ,WAAW,CAAC,CAAC,CAAC,CACb,8BAAC,uBAAU,qBAAC,QAAQ,EAAE,WAAW,CAAC,QAAQ,IAAM,WAAW,IAAE,SAAS,EAAE,OAAO,CAAC,MAAM;YACpF,8BAAC,WAAI,IAAC,IAAI,EAAC,aAAa,GAAG,CAChB,CACd,CAAC,CAAC,CAAC,SAAS,CACT,CACP,CAAC;AACJ,CAAC,CACF,CAAC"}