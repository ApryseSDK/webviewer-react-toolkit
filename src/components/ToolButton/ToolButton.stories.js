'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importDefault(require('react'));
var action_1 = require('../../storybook-helpers/action/action');
var Icon_1 = require('../Icon');
var ToolButton_1 = require('../ToolButton');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/ToolButton',
  component: ToolButton_1.ToolButton,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(
    ToolButton_1.ToolButton,
    {
      disabled: (0, addon_knobs_1.boolean)('disabled', false),
      onClick: (0, action_1.action)('onClick'),
      expandProps: (0, addon_knobs_1.boolean)('has expandProps?', false)
        ? {
            position: (0, addon_knobs_1.select)(
              'expandProps.position',
              ['bottom', 'right'],
              'bottom',
            ),
            onClick: (0, action_1.action)('expandProps.onClick'),
          }
        : undefined,
    },
    react_1.default.createElement(Icon_1.Icon, { icon: 'RotateRight' }),
  );
};
exports.Basic = Basic;
//# sourceMappingURL=ToolButton.stories.js.map
