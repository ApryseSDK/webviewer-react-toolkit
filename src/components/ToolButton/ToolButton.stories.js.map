{"version":3,"file":"ToolButton.stories.js","sourceRoot":"","sources":["ToolButton.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAAyD;AACzD,wDAA0B;AAC1B,gEAA+D;AAC/D,gCAA+B;AAC/B,4CAA2C;AAC3C,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,uBAAuB,EAAE,SAAS,EAAE,uBAAU,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAE1F,IAAM,KAAK,GAAG,cAAM,OAAA,CACzB,8BAAC,uBAAU,IACT,QAAQ,EAAE,IAAA,qBAAO,EAAC,UAAU,EAAE,KAAK,CAAC,EACpC,OAAO,EAAE,IAAA,eAAM,EAAC,SAAS,CAAC,EAC1B,WAAW,EACT,IAAA,qBAAO,EAAC,kBAAkB,EAAE,KAAK,CAAC;QAChC,CAAC,CAAC;YACE,QAAQ,EAAE,IAAA,oBAAM,EAAC,sBAAsB,EAAE,CAAC,QAAQ,EAAE,OAAO,CAAC,EAAE,QAAQ,CAAC;YACvE,OAAO,EAAE,IAAA,eAAM,EAAC,qBAAqB,CAAC;SACvC;QACH,CAAC,CAAC,SAAS;IAGf,8BAAC,WAAI,IAAC,IAAI,EAAC,aAAa,GAAG,CAChB,CACd,EAf0B,CAe1B,CAAC;AAfW,QAAA,KAAK,SAehB"}