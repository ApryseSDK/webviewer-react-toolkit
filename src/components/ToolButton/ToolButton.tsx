import classnames from 'classnames';
import React, { ButtonHTMLAttributes, forwardRef } from 'react';
import useAccessibleFocus from '../../hooks/useAccessibleFocus';
import useOnClick from '../../hooks/useOnClick';

export interface ToolButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

export const ToolButton = forwardRef<HTMLButtonElement, ToolButtonProps>(
  ({ className, children, onClick, ...buttonProps }, ref) => {
    const handleOnClick = useOnClick(onClick, { stopPropagation: true });

    const isUserTabbing = useAccessibleFocus();

    const toolButtonClass = classnames(
      'ui__base ui__toolButton',
      {
        'ui__toolButton--disabled': buttonProps.disabled,
        'ui__toolButton--tabbing': isUserTabbing,
      },
      className,
    );

    return (
      <button {...buttonProps} className={toolButtonClass} onClick={handleOnClick} ref={ref}>
        <div className="ui__toolButton__internals">{children}</div>
      </button>
    );
  },
);
