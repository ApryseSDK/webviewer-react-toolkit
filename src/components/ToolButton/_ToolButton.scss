.ui__toolButton {
  background-color: rgba(100, 100, 100, 0.7);
  border: none;
  outline: none;
  height: 32px;
  min-width: 32px;
  padding: $padding-tiny;
  cursor: pointer;
  user-select: none;
  transition: $focus-transition;

  &:hover:not(.ui__toolButton--disabled) {
    background-color: rgba(75, 75, 75, 0.7);
  }

  &:focus:not(.ui__toolButton--disabled) {
    box-shadow: inset 0 0 0 2px rgba(75, 75, 75, 0.7);
  }

  &:active:not(.ui__toolButton--disabled) {
    background-color: rgba(50, 50, 50, 0.7);
  }

  &--disabled {
    cursor: default;

    .ui__toolButton__internals {
      opacity: 0.5;
    }
  }

  &__internals {
    width: 100%;
    height: 100%;
    @include flex-center;
  }
}
