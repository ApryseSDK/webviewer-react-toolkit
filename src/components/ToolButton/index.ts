export { ToolButton as default, ToolButtonProps } from './ToolButton';
