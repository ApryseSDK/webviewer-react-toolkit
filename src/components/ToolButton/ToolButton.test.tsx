import { mount, shallow } from 'enzyme';
import React from 'react';
import { spy } from 'sinon';
import { ToolButton } from '../ToolButton';

describe('ToolButton component', () => {
  it('renders its contents', () => {
    const toolButton = shallow(<ToolButton />);
    expect(toolButton.find('.ui__toolButton')).toHaveLength(1);
  });

  it('snapshot renders default toolButton', () => {
    const toolButton = shallow(<ToolButton />);
    expect(toolButton).toMatchSnapshot();
  });

  it('clicking toolButton triggers onClick prop', () => {
    const onClick = spy();
    mount(<ToolButton onClick={onClick} />)
      .find('button.ui__toolButton__action')
      .simulate('click');
    expect(onClick.callCount).toBe(1);
  });

  it('clicking disabled toolButton does not trigger onClick prop', () => {
    const onClick = spy();
    mount(<ToolButton onClick={onClick} disabled />)
      .find('button.ui__toolButton__action')
      .simulate('click');
    expect(onClick.callCount).toBe(0);
  });
});
