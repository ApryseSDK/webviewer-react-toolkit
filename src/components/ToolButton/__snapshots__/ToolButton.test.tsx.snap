// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolButton component snapshot renders default toolButton 1`] = `
<div
  className="ui__base ui__toolButton ui__toolButton--bottom"
>
  <ForwardRef
    className="ui__toolButton__action ui__toolButton--bottom"
    onClick={[Function]}
  />
</div>
`;
